airflow/www/auth.py

```diff
@@ -35,7 +35,7 @@ def decorated(*args, **kwargs):
             __tracebackhide__ = True  # Hide from pytest traceback.
 
             appbuilder = current_app.appbuilder
-            if not g.user.is_anonymous and not appbuilder.sm.current_user_has_permissions():
+            if not (g.user.is_anonymous or g.user.perms):
```

Comment:
I feel the previous version is a bit easier to understand (i.e. “not anonymous but does not have any permissions”)

```suggestion
            if not g.user.is_anonymous and not g.user.perms:
```

Reply:
See this one @jhtimmins